packages/firefuel/lib/src/firefuel_collection.dart

```diff
@@ -214,31 +232,33 @@ abstract class FirefuelCollection<T extends Serializable>
   }
 
   @override
-  Future<List<T>> where(List<Clause> clauses, {int? limit}) async {
-    final filteredQuery = _queryFromClauses(clauses);
-    final query = limit == null ? filteredQuery : filteredQuery.limit(limit);
+  Future<List<T>> where(
+    List<Clause> clauses, {
+    List<OrderBy>? orderBy,
+    int? limit,
+  }) async {
+    if (clauses.isEmpty) throw MissingValueException(Clause);
+
+    final allOrderBy = _ensureFirstWhereAndOrderBy(clauses, orderBy);
+
+    final query =
+        ref.filter(clauses).sortIfNotNull(allOrderBy).limitIfNotNull(limit);
+
     final snapshot = await query.get();
 
     return snapshot.docs.toListT();
   }
 
-  Query<T?> _queryFromClauses(List<Clause> clauses) {
-    if (clauses.isEmpty) throw MissingValueException(Clause);
+  List<OrderBy>? _ensureFirstWhereAndOrderBy(
+    List<Clause> clauses,
+    List<OrderBy>? orderBy,
+  ) {
+    final firstOrder = OrderBy(field: clauses.first.field);
+
+    final requiresFirstOrderByFromWhere = orderBy != null &&
+        orderBy.isNotEmpty &&
+        orderBy.first.field != firstOrder.field;
 
-    return clauses.fold(ref, (result, clause) {
-      return result.where(
-        clause.field,
-        isEqualTo: clause.isEqualTo,
-        isNotEqualTo: clause.isNotEqualTo,
-        isLessThan: clause.isLessThan,
-        isLessThanOrEqualTo: clause.isLessThanOrEqualTo,
-        isGreaterThan: clause.isGreaterThan,
-        isGreaterThanOrEqualTo: clause.isGreaterThanOrEqualTo,
-        arrayContains: clause.arrayContains,
-        whereIn: clause.whereIn,
-        whereNotIn: clause.whereNotIn,
-        isNull: clause.isNull,
-      );
-    });
+    return requiresFirstOrderByFromWhere ? [firstOrder, ...orderBy!] : orderBy;
```

Comment:
Let's make sure to add a test to see what happens when a user does something like this:

// Pseudo-code
`.where('name').orderBy('age', 'name')`

Our method will add an additional orderBy filter with `name` being first so it would end up like
`.where('name').orderBy('name', 'age', 'name')`

We just need to see what happens with this and figure out if it's acceptable or not.